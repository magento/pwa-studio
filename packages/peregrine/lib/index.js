import * as RestApi from './RestApi';
import * as Util from './util';

// hooks
export { useEventListener } from './hooks/useEventListener';
export { useCarousel } from './hooks/useCarousel';
export { useDropdown } from './hooks/useDropdown';
export { usePagination } from './hooks/usePagination';
export { useRestApi } from './hooks/useRestApi';
export { useRestResponse } from './hooks/useRestResponse';
export { useScrollLock } from './hooks/useScrollLock';
export { useSearchParam } from './hooks/useSearchParam';
export { useSort } from './hooks/useSort';
export { useTypePolicies } from './hooks/useTypePolicies';

export {
    WindowSizeContextProvider,
    useWindowSize
} from './hooks/useWindowSize';
export { getToastId, useToasts, ToastContextProvider } from './Toasts';

// store
export { enhancer, reducers } from './store';

// components
export { default as ContainerChild } from './ContainerChild';
export { default as List, Items, Item } from './List';
export { default as Price } from './Price';
export { default as Router } from './Router';
export {
    default as PeregrineContextProvider
} from './PeregrineContextProvider';

// misc
export { RestApi };
export { Util };
export { default as createTestInstance } from './util/createTestInstance';
