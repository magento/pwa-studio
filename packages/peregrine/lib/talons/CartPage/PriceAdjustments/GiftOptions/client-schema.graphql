# Mocked client fields used for gift receipt and message components
# TODO: As of 2.4.2 these fields now exist, we should remove these and update components.
extend type Cart {
    include_gift_receipt: Boolean!
    include_printed_card: Boolean!
    local_gift_message: String
}
