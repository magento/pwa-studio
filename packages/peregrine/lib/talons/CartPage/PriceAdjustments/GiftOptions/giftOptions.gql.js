import { gql } from '@apollo/client';

/**
 * Local query. GQL support is not available as of today.
 *
 * Once available, we can change the query to match the schema.
 */
const GET_GIFT_OPTIONS = gql`
    query getGiftOptions($cartId: String!) {
        cart(cart_id: $cartId) @client {
            id
            include_gift_receipt
            include_printed_card
            local_gift_message
        }
    }
`;

export default {
    getGiftOptionsQuery: GET_GIFT_OPTIONS
};
