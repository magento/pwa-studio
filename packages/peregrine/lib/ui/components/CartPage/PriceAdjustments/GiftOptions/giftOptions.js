import React from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { Edit2 as EditIcon } from 'react-feather';
import { Form } from 'informed';

import { useGiftOptions } from '@magento/peregrine/lib/talons/CartPage/PriceAdjustments/GiftOptions/useGiftOptions';
import { useStyle } from '@magento/venia-ui/lib/classify';
import Button from '@magento/venia-ui/lib/components/Button';
import Checkbox from '@magento/venia-ui/lib/components/Checkbox';
import Field from '@magento/venia-ui/lib/components/Field';
import FormError from '@magento/venia-ui/lib/components/FormError';
import Icon from '@magento/venia-ui/lib/components/Icon';
import LoadingIndicator, {
    Spinner
} from '@magento/venia-ui/lib/components/LoadingIndicator';
import LinkButton from '@magento/venia-ui/lib/components/LinkButton';
import Price from '@magento/venia-ui/lib/components/Price';
import TextArea from '@magento/venia-ui/lib/components/TextArea';
import TextInput from '@magento/venia-ui/lib/components/TextInput';

import defaultClasses from './giftOptions.module.css';

/**
 * A child component of the PriceAdjustments component.
 * This component displays the form for adding gift options.
 *
 * @param {Object} props
 * @param {Object} [props.classes] CSS className overrides.
 * @param {Object} [props.giftOptionsConfigData] store config data.
 * See [giftOptions.module.css]{@link https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/PriceAdjustments/GiftOptions/giftOptions.module.css}
 * for a list of classes you can override.
 *
 * @returns {React.Element}
 *
 * @example <caption>Importing into your project</caption>
 * import GiftOptions from "@magento/venia-ui/lib/components/CartPage/PriceAdjustments/GiftOptions";
 */
const GiftOptions = props => {
    const { classes: propClasses } = props;
    const {
        loading,
        errors,
        savingOptions,
        giftReceiptProps,
        printedCardProps,
        printedCardPrice,
        giftMessageCheckboxProps,
        giftMessageResult,
        hasGiftMessage,
        showGiftMessageResult,
        cardToProps,
        cardFromProps,
        cardMessageProps,
        editGiftMessageButtonProps,
        cancelGiftMessageButtonProps,
        saveGiftMessageButtonProps,
        optionsFormProps
    } = useGiftOptions();
    const { formatMessage } = useIntl();
    const classes = useStyle(defaultClasses, propClasses);

    if (loading) {
        return <LoadingIndicator />;
    }

    const { allow_order, allow_gift_receipt, allow_printed_card } =
        props?.giftOptionsConfigData || {};

    const includeGiftReceipt =
        allow_gift_receipt === '1' ? (
            <div className={classes.option}>
                <div className={classes.checkboxContainer}>
                    <Checkbox
                        {...giftReceiptProps}
                        data-cy="GiftOptions-includeGiftReceipt"
                        label={formatMessage({
                            id: 'giftOptions.includeGiftReceipt',
                            defaultMessage: 'Include gift receipt'
                        })}
                    />
                </div>

                {savingOptions.includes(giftReceiptProps.field) ? (
                    <div className={classes.savingContainer}>
                        <span className={classes.savingText}>
                            <FormattedMessage
                                id={'giftOptions.saving'}
                                defaultMessage={'Saving'}
                            />
                        </span>
                        <Spinner classes={{ root: classes.savingSpinner }} />
                    </div>
                ) : null}
            </div>
        ) : null;

    const includeGiftMessage =
        allow_order === '1' ? (
            <>
                <div className={classes.option}>
                    <div className={classes.checkboxContainer}>
                        <Checkbox
                            {...giftMessageCheckboxProps}
                            data-cy="GiftOptions-includeGiftMessage"
                            label={formatMessage({
                                id: 'giftOptions.includeGiftMessage',
                                defaultMessage: 'Optional Message'
                            })}
                        />
                    </div>

                    {savingOptions.includes('giftMessage') ? (
                        <div className={classes.savingContainer}>
                            <span className={classes.savingText}>
                                <FormattedMessage
                                    id={'giftOptions.saving'}
                                    defaultMessage={'Saving'}
                                />
                            </span>
                            <Spinner
                                classes={{ root: classes.savingSpinner }}
                            />
                        </div>
                    ) : null}
                </div>

                <div
                    className={
                        showGiftMessageResult
                            ? classes.giftMessageResultContainer
                            : classes.hidden
                    }
                >
                    <div
                        className={classes.giftMessageResult}
                        data-cy="GiftOptions-giftMessageResult"
                    >
                        <p>
                            <FormattedMessage
                                id="giftOptions.giftMessageTo"
                                defaultMessage="<strong>To:</strong> {cardTo}"
                                values={{
                                    cardTo: giftMessageResult.cardTo,
                                    strong: chunks => <strong>{chunks}</strong>
                                }}
                            />
                        </p>
                        <p>
                            <FormattedMessage
                                id="giftOptions.giftMessageFrom"
                                defaultMessage="<strong>From:</strong> {cardFrom}"
                                values={{
                                    cardFrom: giftMessageResult.cardFrom,
                                    strong: chunks => <strong>{chunks}</strong>
                                }}
                            />
                        </p>
                        <p>{giftMessageResult.cardMessage}</p>
                    </div>

                    <LinkButton
                        {...editGiftMessageButtonProps}
                        classes={{ root: classes.editGiftMessageButton }}
                        data-cy="GiftOptions-editGiftMessageButton"
                    >
                        <Icon
                            classes={{ icon: null }}
                            size={16}
                            src={EditIcon}
                        />
                        <span className={classes.actionLabel}>
                            <FormattedMessage
                                id="giftOptions.editGiftMessageButton"
                                defaultMessage="Edit"
                            />
                        </span>
                    </LinkButton>
                </div>

                <div
                    className={
                        !showGiftMessageResult
                            ? classes.giftMessageFields
                            : classes.hidden
                    }
                >
                    <Field
                        id="to"
                        label={formatMessage({
                            id: 'giftOptions.to',
                            defaultMessage: 'To'
                        })}
                    >
                        <TextInput
                            {...cardToProps}
                            data-cy="GiftOptions-cardTo"
                        />
                    </Field>
                    <Field
                        id={cardFromProps.field}
                        label={formatMessage({
                            id: 'giftOptions.from',
                            defaultMessage: 'From'
                        })}
                    >
                        <TextInput
                            {...cardFromProps}
                            data-cy="GiftOptions-cardFrom"
                        />
                    </Field>
                    <Field
                        id="message"
                        label={formatMessage({
                            id: 'giftOptions.message',
                            defaultMessage: 'Message'
                        })}
                    >
                        <TextArea
                            {...cardMessageProps}
                            data-cy="GiftOptions-cardMessage"
                            placeholder={formatMessage({
                                id: 'giftOptions.cardMessage',
                                defaultMessage: 'Enter your message here'
                            })}
                        />
                    </Field>

                    <div className={classes.giftMessageActions}>
                        {hasGiftMessage ? (
                            <Button
                                {...cancelGiftMessageButtonProps}
                                data-cy="GiftOptions-cancelGiftMessageButton"
                            >
                                <FormattedMessage
                                    id="giftOptions.cancelGiftMessageButton"
                                    defaultMessage="Cancel"
                                />
                            </Button>
                        ) : null}
                        <Button
                            {...saveGiftMessageButtonProps}
                            data-cy="GiftOptions-updateGiftMessageButton"
                        >
                            {hasGiftMessage ? (
                                <FormattedMessage
                                    id="giftOptions.updateGiftMessageButton"
                                    defaultMessage="Update Message"
                                />
                            ) : (
                                <FormattedMessage
                                    id="giftOptions.addGiftMessage"
                                    defaultMessage="Add Message"
                                />
                            )}
                        </Button>
                    </div>
                </div>
            </>
        ) : null;

    const includePrintedCard =
        allow_printed_card === '1' ? (
            <div className={classes.option}>
                <div className={classes.checkboxContainer}>
                    <Checkbox
                        {...printedCardProps}
                        data-cy="GiftOptions-includePrintedCard"
                        label={formatMessage(
                            {
                                id: 'giftOptions.includePrintedCard',
                                defaultMessage: 'Add printed card{price}'
                            },
                            {
                                price:
                                    printedCardPrice &&
                                    printedCardPrice.value > 0 ? (
                                        <>
                                            {' ( + '}
                                            <Price
                                                currencyCode={
                                                    printedCardPrice.currency
                                                }
                                                value={printedCardPrice.value}
                                            />
                                            {')'}
                                        </>
                                    ) : null
                            }
                        )}
                    />
                </div>

                {savingOptions.includes(printedCardProps.field) ? (
                    <div className={classes.savingContainer}>
                        <span className={classes.savingText}>
                            <FormattedMessage
                                id={'giftOptions.saving'}
                                defaultMessage={'Saving'}
                            />
                        </span>
                        <Spinner classes={{ root: classes.savingSpinner }} />
                    </div>
                ) : null}
            </div>
        ) : null;

    return (
        <Form
            data-cy="GiftOptions-form"
            {...optionsFormProps}
            className={classes.root}
        >
            <FormError errors={Array.from(errors.values())} />
            {includeGiftReceipt}
            {includeGiftMessage}
            {includePrintedCard}
        </Form>
    );
};

export default GiftOptions;
