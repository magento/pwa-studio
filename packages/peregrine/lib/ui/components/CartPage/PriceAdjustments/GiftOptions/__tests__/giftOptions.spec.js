import React from 'react';
import { IntlProvider } from 'react-intl';

import { createTestInstance } from '@magento/peregrine';
import { useGiftOptions } from '@magento/peregrine/lib/talons/CartPage/PriceAdjustments/GiftOptions/useGiftOptions';

import GiftOptions from '../giftOptions';

jest.mock('react-intl', () => ({
    ...jest.requireActual('react-intl')
}));
jest.mock('informed', () => ({
    Form: ({ children, ...rest }) => <mock-Form {...rest}>{children}</mock-Form>
}));

jest.mock(
    '@magento/peregrine/lib/talons/CartPage/PriceAdjustments/GiftOptions/useGiftOptions'
);
jest.mock('@magento/venia-ui/lib/classify');
jest.mock('@magento/venia-ui/lib/components/Button', () => props => (
    <mock-Button {...props} />
));
jest.mock('@magento/venia-ui/lib/components/Checkbox', () => props => (
    <mock-Checkbox {...props} />
));
jest.mock(
    '@magento/venia-ui/lib/components/Field',
    () => ({ children, ...rest }) => (
        <mock-Field {...rest}>{children}</mock-Field>
    )
);
jest.mock('@magento/venia-ui/lib/components/FormError', () => props => (
    <mock-FormError {...props} />
));
jest.mock('@magento/venia-ui/lib/components/Icon', () => props => (
    <mock-Icon {...props} />
));
jest.mock('@magento/venia-ui/lib/components/LoadingIndicator', () => ({
    __esModule: true,
    default: jest
        .fn()
        .mockImplementation(props => <mock-LoadingIndicator {...props} />),
    Spinner: jest.fn().mockImplementation(props => <mock-Spinner {...props} />)
}));
jest.mock('@magento/venia-ui/lib/components/LinkButton', () => props => (
    <mock-LinkButton {...props} />
));
jest.mock('@magento/venia-ui/lib/components/Price', () => props => (
    <mock-Price {...props} />
));
jest.mock('@magento/venia-ui/lib/components/TextArea', () => props => (
    <mock-TextArea {...props} />
));
jest.mock('@magento/venia-ui/lib/components/TextInput', () => props => (
    <mock-TextInput {...props} />
));

let inputProps = {};

const talonProps = {
    loading: false,
    errors: [],
    savingOptions: [],
    giftReceiptProps: {
        field: 'includeGiftReceipt'
    },
    printedCardProps: {
        field: 'includePrintedCard'
    },
    printedCardPrice: {},
    giftMessageCheckboxProps: {
        field: 'includeGiftMessage'
    },
    giftMessageResult: {},
    hasGiftMessage: false,
    showGiftMessageResult: false,
    cardToProps: {
        field: 'cardTo'
    },
    cardFromProps: {
        field: 'cardFrom'
    },
    cardMessageProps: {
        field: 'cardMessage'
    },
    editGiftMessageButtonProps: {},
    cancelGiftMessageButtonProps: {},
    saveGiftMessageButtonProps: {},
    optionsFormProps: {}
};

const Component = () => {
    return (
        <IntlProvider locale="en-US">
            <GiftOptions {...inputProps} />
        </IntlProvider>
    );
};

const givenDefaultValues = () => {
    inputProps = {
        giftOptionsConfigData: null
    };
};

const givenOptionsConfigData = () => {
    inputProps.giftOptionsConfigData = {
        allow_order: '1',
        allow_gift_receipt: '1',
        allow_printed_card: '1'
    };
};

describe('#GiftOptions', () => {
    beforeEach(() => {
        givenDefaultValues();
    });

    it('renders loading indicator when loading', () => {
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            loading: true
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders empty form when no options are enabled', () => {
        useGiftOptions.mockReturnValueOnce(talonProps);

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders form when options are enabled', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce(talonProps);

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders form returns errors', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            errors: new Map([['query', { message: 'error' }]])
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders form when gift message form is opened', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            showGiftMessageResult: true
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders form when gift message exists', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            hasGiftMessage: true,
            giftMessageResult: {
                cardTo: 'To',
                cardFrom: 'From',
                cardMessage: 'Message'
            }
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders form when gift message exists and form is opened', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            hasGiftMessage: true,
            giftMessageResult: {
                cardTo: 'To',
                cardFrom: 'From',
                cardMessage: 'Message'
            },
            showGiftMessageResult: true
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders printed card price when above 0', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            printedCardPrice: {
                currency: 'USD',
                value: 10
            }
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders spinners when gift options are saving', () => {
        givenOptionsConfigData();
        useGiftOptions.mockReturnValueOnce({
            ...talonProps,
            savingOptions: [
                'includeGiftReceipt',
                'includePrintedCard',
                'giftMessage'
            ]
        });

        const tree = createTestInstance(<Component />);

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
