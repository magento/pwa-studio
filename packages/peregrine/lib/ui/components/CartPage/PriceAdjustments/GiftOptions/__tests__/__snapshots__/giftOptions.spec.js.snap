// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#GiftOptions renders empty form when no options are enabled 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
</mock-Form>
`;

exports[`#GiftOptions renders form returns errors 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={
      Array [
        Object {
          "message": "error",
        },
      ]
    }
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         
      </p>
      <p>
        <strong>
          From:
        </strong>
         
      </p>
      <p />
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="giftMessageFields"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Add Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label="Add printed card"
      />
    </div>
  </div>
</mock-Form>
`;

exports[`#GiftOptions renders form when gift message exists 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         To
      </p>
      <p>
        <strong>
          From:
        </strong>
         From
      </p>
      <p>
        Message
      </p>
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="giftMessageFields"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Cancel
      </mock-Button>
      <mock-Button>
        Update Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label="Add printed card"
      />
    </div>
  </div>
</mock-Form>
`;

exports[`#GiftOptions renders form when gift message exists and form is opened 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
  </div>
  <div
    className="giftMessageResultContainer"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         To
      </p>
      <p>
        <strong>
          From:
        </strong>
         From
      </p>
      <p>
        Message
      </p>
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="hidden"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Cancel
      </mock-Button>
      <mock-Button>
        Update Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label="Add printed card"
      />
    </div>
  </div>
</mock-Form>
`;

exports[`#GiftOptions renders form when gift message form is opened 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
  </div>
  <div
    className="giftMessageResultContainer"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         
      </p>
      <p>
        <strong>
          From:
        </strong>
         
      </p>
      <p />
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="hidden"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Add Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label="Add printed card"
      />
    </div>
  </div>
</mock-Form>
`;

exports[`#GiftOptions renders form when options are enabled 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         
      </p>
      <p>
        <strong>
          From:
        </strong>
         
      </p>
      <p />
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="giftMessageFields"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Add Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label="Add printed card"
      />
    </div>
  </div>
</mock-Form>
`;

exports[`#GiftOptions renders loading indicator when loading 1`] = `<mock-LoadingIndicator />`;

exports[`#GiftOptions renders printed card price when above 0 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         
      </p>
      <p>
        <strong>
          From:
        </strong>
         
      </p>
      <p />
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="giftMessageFields"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Add Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label={
          Array [
            "Add printed card",
            <React.Fragment>
               ( + 
              <Unknown
                currencyCode="USD"
                value={10}
              />
              )
            </React.Fragment>,
          ]
        }
      />
    </div>
  </div>
</mock-Form>
`;

exports[`#GiftOptions renders spinners when gift options are saving 1`] = `
<mock-Form
  className="root"
>
  <mock-FormError
    errors={Array []}
  />
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftReceipt"
        label="Include gift receipt"
      />
    </div>
    <div
      className="savingContainer"
    >
      <span
        className="savingText"
      >
        Saving
      </span>
      <mock-Spinner
        classes={
          Object {
            "root": "savingSpinner",
          }
        }
      />
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includeGiftMessage"
        label="Optional Message"
      />
    </div>
    <div
      className="savingContainer"
    >
      <span
        className="savingText"
      >
        Saving
      </span>
      <mock-Spinner
        classes={
          Object {
            "root": "savingSpinner",
          }
        }
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <div
      className="giftMessageResult"
    >
      <p>
        <strong>
          To:
        </strong>
         
      </p>
      <p>
        <strong>
          From:
        </strong>
         
      </p>
      <p />
    </div>
    <mock-LinkButton
      classes={
        Object {
          "root": "editGiftMessageButton",
        }
      }
    >
      <mock-Icon
        classes={
          Object {
            "icon": null,
          }
        }
        size={16}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "color": [Function],
              "size": [Function],
            },
            "render": [Function],
          }
        }
      />
      <span
        className="actionLabel"
      >
        Edit
      </span>
    </mock-LinkButton>
  </div>
  <div
    className="giftMessageFields"
  >
    <mock-Field
      id="to"
      label="To"
    >
      <mock-TextInput
        field="cardTo"
      />
    </mock-Field>
    <mock-Field
      id="cardFrom"
      label="From"
    >
      <mock-TextInput
        field="cardFrom"
      />
    </mock-Field>
    <mock-Field
      id="message"
      label="Message"
    >
      <mock-TextArea
        field="cardMessage"
        placeholder="Enter your message here"
      />
    </mock-Field>
    <div
      className="giftMessageActions"
    >
      <mock-Button>
        Add Message
      </mock-Button>
    </div>
  </div>
  <div
    className="option"
  >
    <div
      className="checkboxContainer"
    >
      <mock-Checkbox
        field="includePrintedCard"
        label="Add printed card"
      />
    </div>
    <div
      className="savingContainer"
    >
      <span
        className="savingText"
      >
        Saving
      </span>
      <mock-Spinner
        classes={
          Object {
            "root": "savingSpinner",
          }
        }
      />
    </div>
  </div>
</mock-Form>
`;
